tiledb/sm/array_schema/test/unit_domain_data.cc

```diff
@@ -0,0 +1,144 @@
+/**
+ * @file tiledb/sm/array_schema/test/unit_domain_data.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines a test `main()`
+ */
+
+#include <catch.hpp>
+#include "../domain_typed_data_view.h"
+#include "../dynamic_array.h"
+/*
+ * Instantiating the class `Domain` requires a full definition of `Dimension` so
+ * that its destructor is visible. The need to include this header indicates
+ * some kind of trouble with definition order that needs to be fixed.
+ */
+#include "tiledb/sm/array_schema/dimension.h"
+
+using namespace tiledb::common;
+using namespace tiledb::sm;
+
+/**
+ * Null initializer takes any argument, so it's suitable for both
+ * DynamicArrayStorage and for DomainTypedDataView.
+ */
+struct NullInitializer {
+  template <class... Args>
+  inline static void initialize(Args&&...) {
+  }
+};
+
+TEST_CASE("DynamicArray::DynamicArray, no initializer") {
```

Comment:
`DynamicArray` doesn't belong in the directory at all. It deserves to be given the standard library container treatment so that can be used independently. I'm saving all that work for a future PR.

Reply:
Resolving